caluma/caluma_form/jexl.py

```diff
@@ -67,12 +70,49 @@ def _question(self, slug):
             )
         return question
 
+    def _all_containers_hidden(self, question):
+        """Check whether all containers of the given question are hidden.
+
+        The question could be used in multiple sub-forms in the given
+        document structure. This goes through all paths from the main form
+        to the given question, and checks if they're hidden.
+
+        If all subforms are hidden where the question shows up,
+        the question shall be hidden as well.
+        """
+
+        doc_form = self.context.get("document").form
+
+        # first, build a list of paths to the question
+        def find_path(fq, acc):
```

Comment:
nitpick: I don't know what's meant by `fq` - maybe rather use a longer variable name?

Reply:
`question` would be the obvious choice, but that's already present in the outer context. I'll think of something